lib/html_parser.dart

```diff
@@ -766,20 +774,20 @@ class StyledText extends StatelessWidget {
   final InlineSpan textSpan;
   final Style style;
   final double textScaleFactor;
+  final RenderContext renderContext;
 
   const StyledText({
     this.textSpan,
     this.style,
     this.textScaleFactor = 1.0,
+    this.renderContext,
   });
 
   @override
   Widget build(BuildContext context) {
     return SizedBox(
-      width:
-          style.display == Display.BLOCK || style.display == Display.LIST_ITEM
-              ? double.infinity
-              : null,
+      width: (style.display == Display.BLOCK || style.display == Display.LIST_ITEM) && renderContext.parser.shrinkWrap != true
+        ? double.infinity : renderContext.parser.shrinkWrap == true ? MediaQuery.of(renderContext.buildContext).size.width : null,
```

Comment:
Even though I just commented above on expression simplification, I personally dislike the look of a nested ternary. I would pull this out into some sort of `calculateWidth` method (one ternary is fine, just the nested one needs to go)